javascript/npm/draco3d/draco_nodejs_example.js

```diff
@@ -82,81 +84,37 @@ function encodeMeshToFile(mesh, decoder) {
-
-  const normalAttId = decoder.GetAttributeId(mesh, decoderModule.NORMAL);
-  if (normalAttId > -1) {
-    console.log("Adding normal attribute.");
-    const norAttribute = decoder.GetAttribute(mesh, normalAttId);
-    const norAttributeData = new decoderModule.DracoFloat32Array();
-    decoder.GetAttributeFloatForAllPoints(mesh, norAttribute,
-        norAttributeData);
-    const normals = new Float32Array(numVertexCoord);
-    for (let i = 0; i < numVertexCoord; i += 3) {
-      normals[i] = norAttributeData.GetValue(i);
-      normals[i + 1] = norAttributeData.GetValue(i + 1);
-      normals[i + 2] = norAttributeData.GetValue(i + 2);
-    }
-    decoderModule.destroy(norAttributeData);
-    meshBuilder.AddFloatAttributeToMesh(newMesh, encoderModule.NORMAL,
-        numPoints, 3, normals);
-  }
+  const attrs = {POSITION: 3, NORMAL: 3, COLOR: 3, TEX_COORD: 2};
+
+  Object.keys(attrs).forEach((attr) => {
+    const stride = attrs[attr];
+    const decoderAttr = decoderModule[attr];
+    const encoderAttr = decoderModule[attr];
+    const attrId = decoder.GetAttributeId(mesh, decoderAttr);
 
-  const texAttId = decoder.GetAttributeId(mesh, decoderModule.TEX_COORD);
-  if (texAttId > -1) {
-    const texAttribute = decoder.GetAttribute(mesh, texAttId);
-    const texAttributeData = new decoderModule.DracoFloat32Array();
-    decoder.GetAttributeFloatForAllPoints(mesh, texAttribute,
-        texAttributeData);
-    assertEquals(numVertexCoord, texAttributeData.size());
-    const texcoords = new Float32Array(numVertexCoord);
-    for (let i = 0; i < numVertexCoord; i += 3) {
-      texcoords[i] = texAttributeData.GetValue(i);
-      texcoords[i + 1] = texAttributeData.GetValue(i + 1);
-      texcoords[i + 2] = texAttributeData.GetValue(i + 2);
+    if (attrId < 0) {
+      return;
     }
-    decoderModule.destroy(texAttributeData);
-    meshBuilder.AddFloatAttributeToMesh(newMesh, encoderModule.TEX_COORD,
-        numPoints, 3, normals);
-  }
 
-  const colorAttId = decoder.GetAttributeId(mesh, decoderModule.COLOR);
-  if (colorAttId > -1) {
-    const colAttribute = decoder.GetAttribute(mesh, colorAttId);
-    const colAttributeData = new decoderModule.DracoFloat32Array();
-    decoder.GetAttributeFloatForAllPoints(mesh, colAttribute,
-        colAttributeData);
-    assertEquals(numVertexCoord, colAttributeData.size());
-    const colors = new Float32Array(numVertexCoord);
-    for (let i = 0; i < numVertexCoord; i += 3) {
-      colors[i] = colAttributeData.GetValue(i);
-      colors[i + 1] = colAttributeData.GetValue(i + 1);
-      colors[i + 2] = colAttributeData.GetValue(i + 2);
+    console.log("Adding %s attribute", attr);
+
+    const attribute = decoder.GetAttribute(mesh, attrId);
+    const attributeData = new decoderModule.DracoFloat32Array();
+    decoder.GetAttributeFloatForAllPoints(mesh, attribute, attributeData);
+
+    assert(numVertexCoord === attributeData.size(), 'Wrong attribute size.');
```

Comment:
Is this going to work for texture coordinates? 
`numVertexCoords` is set to be `numPoints * 3`  , but texture coordinates have only `numPoints * 2` values so it would probably fail .. maybe 
assert(numPoints * stride === attributeData.size())